### PR TITLE
Refactor bufprint package

### PR DESCRIPTION
**Pull request by oliversun9**
Printing stuff in `buf beta registry ...` can be put into three categories:
* Printing a page (with a page token)
* Printing the info of an entity (table form in `text` and detailed form in `json`)
* Printing the name of entities (full name in `text` and detailed from in `json`)

This PR replaces `[Commit,Label,Module,Organization]Printer` with three print functions.

This also adds `bufprint` struct tag to the `output[Commit/Label/Module/Organization]` types, but this tag is limited within the `bufprint` package.

`CuratedPluginPrinter` and `TokenPrinter` are left alone -- they can be updated/removed when we work on commands that use these printers.

### CODE REVIEW

**Comment by bufdev** [comment]
This has been in draft for a bit, is this ready to go?

**Comment by oliversun9** [comment]
> This has been in draft for a bit, is this ready to go?

Yep it should be ready as soon as the base branch gets merged. However, it seems there is a tiny merge conflict, I will resolve that and make it non-draft.

File: private/buf/bufprint/bufprint.go
Comment (bufdev) [inline private/buf/bufprint/bufprint.go:91]: `PrintNames`?

Comment (oliversun9) [reply private/buf/bufprint/bufprint.go]: resolved in [785f89b](https://github.com/bufbuild/buf/pull/3173/commits/785f89b564deb9d4752363a922a70c7bdebdb572)

File: private/buf/bufprint/bufprint.go
Comment (oliversun9) [reply private/buf/bufprint/bufprint.go]: fixed in [f2c9991](https://github.com/bufbuild/buf/pull/3173/commits/f2c99916a6e9a70b4931da4543fe7f095ecbde60)

File: private/buf/bufcli/flags_args.go
Comment (bufdev) [inline private/buf/bufcli/flags_args.go:31]: Needs comment

File: private/buf/bufcli/flags_args.go
Comment (bufdev) [inline private/buf/bufcli/flags_args.go:147]: Why do we need emptyDefault?

File: private/buf/bufprint/bufprint.go
Comment (bufdev) [inline private/buf/bufprint/bufprint.go:87]: Entity

File: private/buf/bufprint/bufprint.go
Comment (bufdev) [inline private/buf/bufprint/bufprint.go:178]: PrintEntity

File: private/buf/bufprint/bufprint.go
Comment (bufdev) [inline private/buf/bufprint/bufprint.go:200]: NewLabelEntity (and same downwards, suffix with Entity)

**Comment by bufdev** [comment]
Note to self: reviewed, waiting on @oliversun9 feedback

File: private/buf/bufprint/bufprint.go
Comment (oliversun9) [reply private/buf/bufprint/bufprint.go]: fixed in [1aac338](https://github.com/bufbuild/buf/pull/3173/commits/1aac338032f18da66e0da1540f399a51092a5a0f)

File: private/buf/bufprint/bufprint.go
Comment (oliversun9) [reply private/buf/bufprint/bufprint.go]: fixed in [c5b6335](https://github.com/bufbuild/buf/pull/3173/commits/c5b6335f472fee63a68d1c10ac32c12d1ee76b77)

**Review by bufdev — APPROVED** [review]
Do a QA pass, if it passes, good to merge

**Pull request merged**
